aspnetcore/host-and-deploy/azure-apps/troubleshoot.md

```diff
@@ -5,12 +5,203 @@ description: Learn how to diagnose problems with ASP.NET Core Azure App Service
+
+## App startup errors
+
+**502.5 Process Failure**  
+The worker process fails. The app doesn't start.
+
+This common startup error occurs when the [ASP.NET Core Module](xref:fundamentals/servers/aspnet-core-module) attempts to start the worker process to host the app but it fails to start. For example, this error occurs when the app's assembly is misnamed in the *web.config* file. If an app's assembly is named *trobleshoot.dll* but the *web.config* calls for `arguments="troubleshoo.dll"` (missing the last "t"), the worker process can't be started given the incorrect assembly file name. Examining the Application Event Log often helps troubleshoot this type of problem. Accessing the log is explained later in this topic.
+
+The *502.5 Process Failure* error page is returned when a misconfigured app causes the worker process to fail:
+
+![Browser window showing the 502.5 Process Failure page](troubleshoot/_static/process-failure-page.png)
+
+**500 Internal Server Error**  
+The app starts, but an error prevents the server from fulfilling the request.
+
+When an error occurs within the app's code during startup or creating a response, the app may produce an invalid response (for example, the response contains no content) or a *500 Internal Server Error* message in the browser. The Application Event Log usually states that the app started normally and is listening on a given port. From the server's perspective that's correct. It's the app itself which is unhealthy. Running the app in the [Kudu](https://github.com/projectkudu/kudu/wiki) console on the server or enabling the ASP.NET Core Module stdout log usually helps troubleshoot the problem. Running the app in the Kudu console and accessing the stdout log is described later in this topic.
+
+When the app is in its Production environment, the browser displays a friendly error page for 500-series errors:
+
+![Browser window showing the 500 Internal Server Error page. The page presents the friendly error message 'An error occurred while starting the application.'](troubleshoot/_static/internal-server-error.png)
+
+## Troubleshoot app startup errors
+
+### Application Event Log
+
+Use the **Diagnose and solve problems** blade in the Azure portal to access the Application Event Log:
+
+1. In the Azure portal, open the app's blade in the **App Services** blade.
+1. Select the **Diagnose and solve problems** blade.
+1. Under **SELECT PROBLEM CATEGORY**, select the **Web App Down** button.
+1. Under **Suggested Solutions**, open the pane for **Open Application Event Logs**. Select the **Open Application Event Logs** button.
+1. Examine the latest error provided by the *IIS AspNetCoreModule* in the **Source** column.
+
+For example, the log displays an error:
+
+![Browser window of the Application Events Log showing an error message from the IIS AspNetCoreModule](troubleshoot/_static/app-event-log-error.png)
```

Comment:
I wouldn't call it "very" rare since devs mess with those settings all the time and sometimes don't use sdkweb targets by mistake.

I wanted a perfectly clear and understandable error here as an example because I don't want the reader to spend time thinking about what the error is when I'm making more important points.

If someone has the wrong assembly name, then of course the process can't start because that assembly doesn't exist. The reader isn't going to focus or dwell on that example. They should remain focused on the more important points of what one of these 502.5 looks like in the App Event Log and how it phrases the problem of 'I can't start the process' in its message.

If you still want it changed, let me know the error condition you'd like there, and I'll swap it in. I just hope we don't have something here that will cause the reader to stop for 15 seconds to think about.

Reply:
throw("The seed method failed"); or something like that.

Throw in main. You get that exact error message.

Reply:
I think my error message is much more clear.

Reply:
BTW, in my secure user data sample, I've had 4 customers report failure on azure. The seed method failed because they didn't use a strong password. The sample now throws on that failure.

Reply:
Isn't that approach going away with their new bits for 2.1? ... https://github.com/aspnet/EntityFrameworkCore/issues/629

Reply:
Has nothing to do with Seeding directly, it's creating the users that fails.

Reply:
oh ... I didn't even look at it beyond this ...

```csharp
public class BloggingContextSeedData : DatabaseSeedData<BloggingContext>
{
    public override void Seed(BloggingContext context)
    { 
    }
}
```
... and that looked like they were going to provide a method for seeding.